fatips/1.md

```diff
@@ -6,221 +6,252 @@
-| tokens[n]             | object | The token object                                             |                                                              | N        |
-| tokens[n].id          | string | The unique ID of the token                                   | length > 0. Unique among other tokens within this asset      | Y        |
-| tokens[n].salt        | string | Random string to be hashed into token idNonce(sha256d(tokenID + salt)) |                                                              | Y        |
-| tokens[n].idSignature | string | ed25519 signature from issuing identity's SK1 key            | ed25519 signature validation (idKey, idNonce, idSignature)   | Y        |
-| tokens[n].metadata    | any    | Optional user defined metadata JSON object                   |                                                              | N        |
-|                       |        |                                                              |                                                              |          |
-| salt                  | string | Random string to be hashed into nonce to be signed.          | user defined. Optional                                       | Y        |
-| idSignature           | string | ed25519 signature from issuing identity's SK1 key            | ed25519 signature validation (idKey, sha256d(tokenID + salt), signature) | Y        |
-
+| Name               | Type   | Description                                                  | Validation                                | Required |
+| ------------------ | ------ | ------------------------------------------------------------ | ----------------------------------------- | -------- |
+| type               | string | The type of this token issuance                              | Must equal 'FAT-0'                        | Y        |
+| supply             | number | The maximum possible number of tokens that can be in circulation. If omitted the supply is infinite. | Must be > 0                               | N        |
+| name               | string | The display name of the token                                |                                           | N        |
+| symbol             | string | The display symbol of the token                              | must be A-Z, and 1-4 characters in length | N        |
+| tokens             | array  | An array of token objects, representing tokens being issued  | Must be an array of objects               | N        |
+| tokens[n].id       | string | The unique ID of the token being issued                      | Length > 0                                | Y        |
+| tokens[n].metadata | object | Object holding user defined key-value pairs                  |                                           | N        |
+| salt               | string | Random string to salt the issuance                           | user defined. Optional                    | N        |
```

Comment:
I'm not sure the salt is ever needed here either. Those entries are "naturally salted" by the token IDs. Nobody could recreate the same issuance (either voluntarily or maliciously) because it would be discarded by the fact that those token ids have already been issued. In which case that salt would be useful?

Reply:
@PaulBernier this is a good point. However, it is not currently required that `tokens` be present in the first issuance entry.

@AdamSLevy In FAT-0 I believe we can get away with that because there is a single authoritative issuance entry that establishes the token(we don't have to worry about replay vulnerabilities on the same chain). In FAT-1 issuance entries are ongoing, as I so in-eloquently stated in the spec :stuck_out_tongue: 

>  Token issuance entries are ongoing like transaction entries to allow ongoing issuance of tokens by the issuer.

There is a risk of replay, albeit the replay would currently be ignored/invalid since it would be attempting to re-issue already issued tokens. However, If we add additional functionality to the issuance entries (for example FATIP-201 for address workflows) this could be an attack vector people could use to re-enable their addresses, and all other types of same chain griefery. In terms of security & protection, we should approach these FATIP-1 issuance entries more like transactions. It may be necessary to use `blockheight` and institute validation like in transactions.